AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/DefaultKeyStorageInterface.dfy

```diff
@@ -793,155 +806,102 @@ module DefaultKeyStorageInterface {
-            "Table Name: " + ddbTableName + "\tBranch Key ID: " + input.Identifier));
-      var itemList: DDB.ItemList := ddbResponse.Responses.value[ddbTableName];
-      :- Need(
-        2 <= |itemList| <= 3,
-        Types.KeyStorageException(
-          message:="DDB returned the wrong number of Items;"
-          + " 2 or 3 are expected. Received: " + String.Base10Int2String(|itemList|)));
+            "GetItemsForInitializeMutation: No items returned. "
+            + "Branch Key ID: " + input.Identifier
+            + "\tTable Name: " + ddbTableName));
 
       /** Process sensical DDB Response */
-      var itemIndex := 0;
-      var beaconItem: Option<Types.EncryptedHierarchicalKey> := None;
-      var activeItem: Option<Types.EncryptedHierarchicalKey> := None;
-      var lockItem: Option<Types.MutationLock> := None;
-      while itemIndex < |itemList|
-      {
-        var item := itemList[itemIndex];
-        :- Need(
-          Structure.TYPE_FIELD in item.Keys,
-          Types.KeyStorageException(
-            message:=
-              "Malformed Branch Key Store Item encountered. Missing TYPE Attribute. "
-              + "TableName: " + ddbTableName +
-              "\tBranch Key ID: " + input.Identifier
-          ));
-        var itemType := item[Structure.TYPE_FIELD];
-        :- Need(
-          itemType.S?,
-          Types.KeyStorageException(
-            message:=
-              "Malformed Branch Key Store Item encountered. TYPE Attribute MUST be String but is not. " +
-              "TableName: " + ddbTableName + "\tBranch Key ID: " + input.Identifier
-          ));
-        var typeStr: string := itemType.S;
-        var id: string := input.Identifier; // Tony did this solely so lines would not wrap
-        match typeStr {
-          case MUTATION_LOCK_TYPE =>
-            var lockItem? := mutationLockFromItem(item, id);
-            if (lockItem?.Success?) { lockItem := Some(lockItem?.value); } else { return Failure(lockItem?.error); }
-          case BRANCH_KEY_ACTIVE_TYPE => var activeItem? := encryptedHierarchicalKeyFromItem(item, logicalKeyStoreName, id);
-          if (activeItem?.Success?) { activeItem := Some(activeItem?.value); } else { return Failure(activeItem?.error); }
-          case BEACON_KEY_TYPE_VALUE => var beaconItem? := encryptedHierarchicalKeyFromItem(item, logicalKeyStoreName, id);
-          if (beaconItem?.Success?) { beaconItem := Some(beaconItem?.value); } else { return Failure(beaconItem?.error); }
-          case _ => return Failure(
-              Types.KeyStorageException(
-                message:=
-                  "Unexpected item returned by DDB. TableName: "
-                  + ddbTableName + "\tBranch Key ID: " + input.Identifier));
-        }
-        itemIndex := 1 + itemIndex;
-      }
+      var lockItem: Option<Types.MutationLock> :-
+        MutationLockFromOptionalItem(ddbResponse.Responses.value[0].Item, input.Identifier);
```

Comment:
imo; it is easier to reason about which response item you are handling if you break them out beforehand - it is more legible and I don't have to think too hard. future you would be thankful

```
var x := ddbResponse.Responses.value[0];
var y := ddbResponse.Responses.value[1];
...
```

Reply:
I can refactor this.
I even had it written this way,
but an earlier PR encouraged me to not have variables allocated that were not needed.

I took that feedback too far; 
I agree that it would be a cleaner read to identify the three items.

Question: Is the cleanest implementation to assign each type an integer?

Because then we read:
```
MutationLockFromOptionalItem(ddbResponse.Responses.value[mLockIndex].Item, input.Identifier);
```